### DISCUSSION TITLE
Home Assistant su Raspberry e Starlink

### DISCUSSION THREAD

**Discussion by vebto53** [indomus/forum#3074] [Networking]
Ciao a tutti, ho in mente di prendere Starlink (vers.base) (la mia zona è morta per le comunicazioni). Mi sto preoccupando di poter accedere da remoto a Home Assistant, poichè ho letto che starlink usa una sua rete privata e non da, per la versione base, un IP pubblico e i vari DDNS non funzionano. Qualcuno ha già fatto questa esperienza e ha risolto in qualche modo?
Grazie

**Comment by snakuzzo** [comment]
Problema risolvibile utilizzando il tunnel Cloudflared Zero Trust. Devi solo investire pochi euro l'anno per comprare un tuo dominio.
Ad oggi, noi consigliamo a prescindere questa soluzione (anche se si ha un IP pubblico) anche per tutta una serie di opzioni e funzionalità legate alla sicurezza.

Di seguito le due guide che ti interesseranno...

- https://indomus.it/guide/configurare-cloudflare-per-il-proprio-dominio-personale-per-protezione-reverse-proxy-cdn-e-altro/
- https://indomus.it/guide/collegarsi-da-remoto-a-home-assistant-via-reti-provider-tradizionali-o-che-usino-nat-via-cloudflare/

**Reply by vebto53** [reply]
Scusa il ritardo, ma ho dovuto fare con calma. Ti ringrazio ho seguito le guide e sono riuscito ad accedere con dominio e cloudflare. Sono ancora con la vecchia adsl, perché Starlink non è ancora arrivato. Appena posso provo, ma non ho dubbi che funzionerà.
Non so se devo mettere "risolto" da qualche parte. Grazie

**Reply by snakuzzo** [reply]
Ciao. Quando cambierai connettività ti basterà avere accesso ad internet e tutto funzionerà come sta funzionando adesso.

Non preoccuparti, non c'è bisogno di indicare come risolto.

**Reply by vebto53** [reply]
Ciao, non sono pratico del forum, ma a mia problematica relativa a Alexa e starlink con HA mi hanno rimandato su questa conversazione. Presumo ti abbiano passato la qst. Attendo Grazie intanto

**Reply by snakuzzo** [reply]
Sì, ma se non indichi cosa ti serve nello specifico, uno come fa a darti una mano ? :D
Comunque, ho trovato la [tua discussione](https://github.com/indomus/forum/discussions/3088).

Se avevi configurato la lambda di AWS con il dominio vecchio, devi:

- aggiornare il config.json sulla lambda con il nuovo dominio:nuova porta
- deployare
- fare il test

E per magia dovrebbe rifunzionare tutto.

**Reply by vebto53** [reply]
Grazie e scusa per il disguido, ma mi son trovato la discussione che linkava a questa e pensavo ci fosse un collegamento. Adesso so come fare.Saluti. Pietro

**Reply by vebto53** [reply]
Ancora a chiedere un aiuto. Ho modificato il config.json sostituendo il dominio, mettendo come porta la 80. Deploy ok 
Faccio il test (ho camuffato il dominio):
```
  "errorMessage": "405 Client Error: Method Not Allowed for url: https://xxxxxcca.top/api/alexa/smart_home",
  "errorType": "HTTPError",
  "requestId": "03638505-7bab-44e4-8fda-e4a488bdb94f",
  "stackTrace": [
    "  File \"/var/task/haaska.py\", line 111, in event_handler\n    return ha.post('alexa/smart_home', event, wait=True)\n",
    "  File \"/var/task/haaska.py\", line 66, in post\n    r.raise_for_status()\n",
    "  File \"/var/task/requests/models.py\", line 943, in raise_for_status\n    raise HTTPError(http_error_msg, response=self)\n"
  ]
```
Cosa ho sbagliato?
Grazie

**Reply by vebto53** [reply]
Questo è il config.json
```
{
  "url": "http://xxxxxcca.top:80/api",
  "bearer_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJmMDI5OTBhMDc5YTY0NzhjOThlODE5NDIzNDU1YTA3ZiIsImlhdCI6MTYyMTg2NTY5OCwiZXhwIjoxOTM3MjI1Njk4fQ.2X_eAflKDCAYqoXqHkaMdmN4CvhYDXKbqWVDPp2XZ74",
  "debug": false,
  "ssl_verify": true,
  "ssl_client": []
}
```

**Reply by snakuzzo** [reply]
hai la URL in http, non funzionerà mai.
La URL da usare deve essere `https://tuodominio:tuaporta/api` (se `tuaporta` è la `443` puoi anche ometterla)

**Comment by vebto53** [comment]
Non so quale porta mettere. Dunque ho connessione con starlink per accedere da remoto ho seguito i tuoi suggerimenti:
- ho preso un dominio su goDaddy (xxxxx.top) 
- poi ho installato e configurato Clouflare 
- a questo punto digitando sul brawser https://xxxxx.top , senza alcuna porta, vado direttamente sulla dashboard di Home Assistant. 
Prima avevo una nat sul mio router che mi rimandava alla porta 8123, ma ora se ho capito bene arrivo diretto con cloudflare.

**Reply by snakuzzo** [reply]
Te l'ho detto prima, mi sembrava di essere stato più che chiaro. Comunque, se raggiungi la dashboard di HA con...

```
https://pippo.top
```

Dovrai mettere

```
https://pippo.top/api
```

**Reply by vebto53** [reply]
L'avevo già provato, ma avevo lasciato http e non https. 
Ora funziona.
Ti ringrazio per la pazienza.
Pietro

**Comment by VoidElle** [comment]
Ciao a tutti, scusate l’intromissione nel thread (dove tra l’altro avete già risolto), ma mi sono imbattuto anch’io in questo problema.

Sono anche io proprietario di Starlink ed ho configurato una soluzione, per l'accesso remoto, utilizzando una VPN chiamata Tailscale, che si integra molto bene con Home Assistant.

Il piano gratuito è limitato a 3 utenti, ma permette l'inserimento di dispositivi illimitati.

L’unico aspetto da considerare è che tutti i dispositivi che vogliono accedere da remoto a Home Assistant devono essere configurati all'utilizzo di questa VPN (operazione comunque molto semplice, basta scaricare il loro programma / app).

**Reply by snakuzzo** [reply]
Dipende da quello che uno vuole, ma permettimi di dire che non c'è proprio confronto tra le due cose. La soluzione con Cloudflare, che di fatto è gratuita, ha tutta una serie di funzionalità aggiuntive che sono veramente il top. L'unico costo che hai è quello del dominio, ma sono "spiccioli"
